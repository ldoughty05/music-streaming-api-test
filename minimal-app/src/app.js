import { Routes, Route } from 'react-router-dom';

import { Counter } from './features/counter/counter.js';

export function App() {
  const page =
    <>
      <Counter label={'Taps'} />
    </>;
  return (
    <Routes>
      <Route path={'/*'} element={page} />
    </Routes>
  );
}
