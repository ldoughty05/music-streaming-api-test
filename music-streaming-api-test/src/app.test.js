import { render } from '@testing-library/react';
import './testing/mockRedux.js';
import { MemoryRouter as Router } from 'react-router-dom';

import { App } from './app.js';

import {
  selectValue,
} from './features/counter/counterSlice.js';
jest.mock('./features/counter/counterSlice.js', () => ({
  selectValue: jest.fn().mockName('selectValue'),
}));

describe('the app', () => {
  test('has one labeled button that displays the value from the store', () => {
    selectValue.mockReturnValue(9999);
    const { container } = render(
      <Router initialEntries={['/']}>
        <App />
      </Router>,
    );
    expect(container).toMatchSnapshot();
  });
});
