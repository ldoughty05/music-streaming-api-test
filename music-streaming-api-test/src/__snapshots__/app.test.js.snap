// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the app has one labeled button that displays the value from the store 1`] = `
<div>
  <label>
    Taps
    :Â 
    <button
      class="blue"
    >
      9999
    </button>
  </label>
</div>
`;
